sdk/core/core-client/src/serviceClient.ts

```diff
@@ -0,0 +1,425 @@
+  HttpsClient,
+  PipelineRequest,
+  PipelineResponse,
+  Pipeline,
+  createPipelineRequest,
+  createPipelineFromOptions,
+  bearerTokenAuthenticationPolicy
+} from "@azure/core-https";
+import {
+  OperationResponse,
+  OperationArguments,
+  OperationSpec,
+  ServiceClientCredentials,
+  OperationRequest,
+  OperationResponseMap,
+  FullOperationResponse,
+  DictionaryMapper,
+  CompositeMapper
+} from "./interfaces";
+import { getPathStringFromParameter, isStreamOperation } from "./interfaceHelpers";
+import { MapperTypeNames } from "./serializer";
+import { getRequestUrl } from "./urlHelpers";
+import { isPrimitiveType } from "./utils";
+import { getOperationArgumentValueFromParameter } from "./operationHelpers";
+import { deserializationPolicy } from "./deserializationPolicy";
+import { signingPolicy } from "./signingPolicy";
+
+/**
+ * Options to be provided while creating the client.
+ */
+export interface ServiceClientOptions {
+  /**
+   * If specified, this is the base URI that requests will be made against for this ServiceClient.
+   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   */
+  baseUri?: string;
+  /**
+   * The default request content type for the service.
+   * Used if no requestContentType is present on an OperationSpec.
+   */
+  requestContentType?: string;
+  /**
+   * Credential used to authenticate the request.
+   */
+  credential?: TokenCredential | ServiceClientCredentials;
+  /**
+   * A customized pipeline to use, otherwise a default one will be created.
+   */
+  pipeline?: Pipeline;
+  /**
+   * The HttpClient that will be used to send HTTP requests.
+   */
+  httpsClient?: HttpsClient;
+
+  /**
+   * A method that is able to turn an XML object model into a string.
+   */
+  stringifyXML?: (obj: any, opts?: { rootName?: string }) => string;
+}
+
+/**
+ * Initializes a new instance of the ServiceClient.
+ */
+export class ServiceClient {
+  /**
+   * If specified, this is the base URI that requests will be made against for this ServiceClient.
+   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   */
+  private readonly _baseUri?: string;
+
+  /**
+   * The default request content type for the service.
+   * Used if no requestContentType is present on an OperationSpec.
+   */
+  private readonly _requestContentType?: string;
+
+  /**
+   * Decoupled method for processing XML into a string.
+   */
+  private readonly _stringifyXML?: (obj: any, opts?: { rootName?: string }) => string;
+
+  /**
+   * The HTTP client that will be used to send requests.
+   */
+  private readonly _httpsClient: HttpsClient;
+
+  private readonly _pipeline: Pipeline;
+
+  /**
+   * The ServiceClient constructor
+   * @constructor
+   * @param credential The credentials used for authentication with the service.
+   * @param options The service client options that govern the behavior of the client.
+   */
+  constructor(options: ServiceClientOptions = {}) {
```

Comment:
I love the new constructor!

Reply:
I tried very hard to get rid of the old cruft. Do you know if we can ditch `ServiceClientCredentials` yet?

Reply:
AppConfiguration library still uses it.  Not sure why their AppConfigurationCredential is not used via a request policy though.

Reply:
Yeah, I think we can move this to using custom policies with `AzureKeyCredential`